cloud-watch-to-slack-testing-58006136-c30b-43d1-880c-7b36f860e7c8/deployment/index.js

```diff
@@ -142,10 +143,38 @@ function sendMessageToSlack(messageText, callback) {
   });
 }
 
+// Set the Slack channel based on the input SNS Topic to Slack Channel map
+// in the snsTopicToSlackChannel env var with format
+// {"<SNS Topic resource id>":"<slack channel name>"}
+// E.g.
+//     {"slack-low-priority-topic":"dockstore-testing",
+//     "slack-medium-priority-topic":"dockstore-dev-alerts",
+//     "slack-high-priority-topic":"dockstore-alerts",
+//     "SSM-to-Slack-SNSTopicToSlack-1VI4KZW1DSADS":"dockstore-dev-testing" }
+// input: SNS topic ARN
+function setSlackChannelBasedOnSNSTopic(topicArn) {
+  // Get the SNS Topic resource ID from the AWS ARN
+  // https://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html
+  const snsTopicResourceID = topicArn.slice(topicArn.lastIndexOf(":") + 1);
+
+  for (let [topicID, channel] of Object.entries(snsTopicToSlackChannelMap)) {
+    console.info(topicID + " = " + channel);
+
+    if (snsTopicResourceID === topicID) {
+      slackChannel = channel;
+      break;
+    }
+  }
+  console.info("Slack channel is " + slackChannel);
+}
+
 function processEvent(event, callback) {
   console.log(event);
   const message = JSON.parse(event.Records[0].Sns.Message);
 
+  const topicArn = event.Records[0].Sns.TopicArn;
```

Comment:
I see we do this pattern already on line 147/163, but it got me thinking and wondering if we're missing any records? Do we know that there is always only one record?

Reply:
this probably dates back to the initial implementation of this, which may have been me.
I saw only one at the time (by inspection! https://en.wikipedia.org/wiki/List_of_mathematical_jargon#Proof_terminology ) but I definitely don't know that there is always only one record

Reply:
There is always only one record for SNS events; however events from other resources could have multiple records, e.g. S3 or DynamoDB, but we don't capture those yet.

Reply:
https://stackoverflow.com/questions/33690231/when-lambda-is-invoked-by-sns-will-there-always-be-just-1-record